File: uplift-gradle-plugins/src/main/java/com/github/kjetilv/uplift/plugins/UpliftDeployTask.java
package com.github.kjetilv.uplift.plugins;

import org.gradle.work.DisableCachingByDefault;

import java.nio.file.Path;
import java.util.List;

/** Deploys the stack, then reports what is now running. */
@DisableCachingByDefault(because = "Deploying a stack is not cacheable")
public abstract class UpliftDeployTask extends UpliftLambdaZipTask {

    @Override
    protected void perform() {
        collectLambdaZips();
        cdk().run("cdk deploy " + profileOption() + " --require-approval=never " + getStack().get());
        List<Path> lambdas = lambdas();
        report(lambdas == null ? List.of() : lambdas);
    }
}

File: uplift-gradle-plugins/src/main/kotlin/com/github/kjetilv/uplift/plugins/UpliftDeployTask.kt
package com.github.kjetilv.uplift.plugins

import org.gradle.process.ExecOperations
import org.gradle.work.DisableCachingByDefault
import javax.inject.Inject

@DisableCachingByDefault(because = "Deploying a stack is not cacheable")
abstract class UpliftDeployTask @Inject constructor(execOperations: ExecOperations) :
    UpliftLambdaZipTask(execOperations) {

    override fun perform() {
        collectLambdaZips()
        deploy()
        ping(lambdas() ?: emptyList())
    }

    private fun deploy() {
        runCdk("cdk deploy ${profileOption()} --require-approval=never ${stack.get()}")
    }
}
